iceoryx_posh/source/roudi/memory/mempool_collection_memory_block.cpp

```diff
@@ -51,8 +51,10 @@ uint64_t MemPoolCollectionMemoryBlock::alignment() const noexcept
 
 void MemPoolCollectionMemoryBlock::onMemoryAvailable(cxx::not_null<void*> memory) noexcept
 {
-    posix::Allocator allocator(memory, size());
-    auto memoryManager = allocator.allocate(sizeof(mepoo::MemoryManager), alignof(mepoo::MemoryManager));
+    BumpAllocator allocator(memory, size());
+    auto allocationResult = allocator.allocate(sizeof(mepoo::MemoryManager), alignof(mepoo::MemoryManager));
+    cxx::Expects(!allocationResult.has_error());
```

Comment:
Same pattern as before. If allocation fails we terminate. This may be tolerable here. It was not checked before and checking is better than continuing without resources.

It would be even better to continue, but this is probably not possible. The next best option is graceful termination, but we lack facilities to do so. This will be corrected soon. :-)

This pattern may occur more often and I will not further comment on it as there is no immediate solution here.

Just an observation, nothing to do for now (and no todo needed as searching for `Expects` is sufficient).

Reply:
Just for info: the `Expects` was hidden inside `Allocator::allocate` before.